src/analyses/ai.h

```diff
@@ -193,37 +212,65 @@ class ai_baset
   }
 
 protected:
-  // overload to add a factory
-  virtual void initialize(const goto_programt &);
-  virtual void initialize(const goto_functionst::goto_functiont &);
-  virtual void initialize(const goto_functionst &);
+  /// Initialize all the domains for a single function. Overloaded this to add a
+  /// factory.
```

Comment:
:pick: I don't know if you know  - but I don't know what "add a factory" means or why I'd want to do it.

Reply:
I don't either - the comment is moved from somewhere else. I guess I should change it to something I understand - I understand what was happening when that comment was added.